noir-projects/noir-contracts/contracts/gas_token_contract/src/main.nr

```diff
@@ -23,6 +23,28 @@ contract GasToken {
         storage.portal_address.initialize(portal_address);
     }
 
+    #[aztec(private)]
+    fn claim(to: AztecAddress, amount: Field, secret: Field) {
+        let content_hash = get_bridge_gas_msg_hash(to, amount);
+        let portal_address = storage.portal_address.read_private();
+        assert(!portal_address.is_zero());
+
+        // Consume message and emit nullifier
+        context.consume_l1_to_l2_message(content_hash, secret, portal_address);
+
+        // TODO(palla/gas) Emit an unencrypted log to announce which L1 to L2 message has been claimed
```

Comment:
Why do we want to do that?

Reply:
Tracing. If the claiming of the L1toL2 message happens in private-land, you lose track of which deposit in L1 corresponds to which claim on L2.